### DISCUSSION TITLE
MM energies for a random molecule?

### DISCUSSION THREAD

**Discussion by Mirrty** [rdkit/rdkit#4817] [Q&A] [answered]
I would like to ask if there is an option of calculating an energy for a specified 3D geometry of a molecule? I have three conformers for a molecule with already calculated x,y,z. I found _MMFFOptimizeMolecule_ and _MMFFOptimizeMoleculeConfs_, but those functions are for conformers generated by _EmbedMolecule_ methods.
Thanks.

**Comment by bp-kelley** [comment, marked as answer]
You can try maxIters=0 to get the initial state I think.

```
print(AllChem.MMFFOptimizeMoleculeConfs(mol, maxIters=0))
```

However, I would be very leery of this, MMFF is very sensitive to bond
lengths and angles, so the initial conformation will be heavily biased by
the starting conformation simply because it was initially fit with a
non-MMFF force field.  I'm not sure RDKIt allows this, but I've had success
minimizing in flat bottomed harmonic wells to "shrug" off such such
starting issues but keep the initial conformations relatively intact for
comparison purposes.

**Comment by ptosco** [comment]
@Mirrty Brian's suggestion is an excellent one; you can achieve the same result constructing a force-field for each conformer and calling its `CalcEnergy()` method:
```
# Brian's suggestion
res = rdForceFieldHelpers.MMFFOptimizeMoleculeConfs(mol, maxIters=0)
[e for _, e in res]
[9.127722026693464,
 35.526021744965675,
 18.042407230661997,
 12.778344118781307,
 11.7500351322213,
 14.144515687294653,
 9.652650968813049,
 18.805206306692167,
 12.849496535548143,
 10.959160478894946]

# CalcEnergy method
mp = rdForceFieldHelpers.MMFFGetMoleculeProperties(mol)
[rdForceFieldHelpers.MMFFGetMoleculeForceField(
    mol, mp, confId=cid).CalcEnergy() for cid in cids]
[9.127722026693464,
 35.526021744965675,
 18.042407230661997,
 12.778344118781307,
 11.7500351322213,
 14.144515687294653,
 9.652650968813049,
 18.805206306692167,
 12.849496535548143,
 10.959160478894946]
```
